lib/checkclass.cpp

```diff
@@ -3438,6 +3444,10 @@ namespace {
     };
 }
 
+#ifdef __clang__
+using CheckClass_internal::MyFileInfo;
```

Comment:
You could use an inline namespace so you don't need the using statement.

Reply:
This is a like 5 line workaround so we do not ship with a crash. All this time discussing this is taking away from the time I could spend to produce an example to actually report upstream. And it prevents me from fixing other things as this blocks other PRs...

Reply:
Thanks for pointing that out. But as I said before - it is just a workaround and we should not overdo it. May initial approach was just bad but I think the current code is okay enough to merge. After all this might just be deleted at some point.

I also wonder if we properly handle inline namespaces in our checks. Yet another rabbit hole to follow.